core/server/worker/src/main/java/alluxio/worker/block/UnderFileSystemBlockReader.java

```diff
@@ -180,10 +180,18 @@ public ByteBuffer read(long offset, long length) throws IOException {
         .checkState(bytesRead == bytesToRead, PreconditionMessage.NOT_ENOUGH_BYTES_READ.toString(),
             bytesRead, bytesToRead, mBlockMeta.getUnderFileSystemPath());
     if (mBlockWriter != null && mBlockWriter.getPosition() < mInStreamPos) {
-      Preconditions.checkState(mBlockWriter.getPosition() >= offset);
-      ByteBuffer buffer = ByteBuffer.wrap(data, (int) (mBlockWriter.getPosition() - offset),
-          (int) (mInStreamPos - mBlockWriter.getPosition()));
-      mBlockWriter.append(buffer.duplicate());
+      try {
+        Preconditions.checkState(mBlockWriter.getPosition() >= offset);
+        mLocalBlockStore.requestSpace(mBlockMeta.getSessionId(), mBlockMeta.getBlockId(),
+            mInStreamPos - mBlockWriter.getPosition());
```

Comment:
Not sure if this is a potential bottleneck or not. But at least for me I think maintaining another state here can be adding unnecessary complexity.